hana3d/src/validators/triangle_count.py

```diff
@@ -0,0 +1,69 @@
+"""Triangle count Validator."""
+
+import logging
+from typing import Tuple
+
+import bpy
+
+from . import BaseValidator, Category
+from ..asset.asset_type import AssetType
+
+MAX_TRIANGLE_COUNT = 100000
+
+
+def _get_triangles_in_mesh(mesh: bpy.types.Mesh) -> int:
+    triangle_count = 0
+    for polygon in mesh.polygons:
+        count = polygon.loop_total
+        if count > 2:
+            triangle_count += count - 2
+    return triangle_count
+
+
+def _get_triangle_count(object_name: str) -> int:
+    triangle_count = 0
+    blend_object = bpy.data.objects[object_name]
+    if blend_object.type == 'MESH':
+        depsgraph = bpy.context.evaluated_depsgraph_get()
+
+        object_eval = blend_object.evaluated_get(depsgraph)
+        mesh_from_eval = bpy.data.meshes.new_from_object(object_eval)
+        triangle_count += _get_triangles_in_mesh(mesh_from_eval)
+        bpy.data.meshes.remove(mesh_from_eval)
+    return triangle_count
+
+
+def _get_object_list(asset_type: AssetType, export_data: dict):
+    if asset_type == AssetType.model:
+        return export_data.get('models', [])
+    elif asset_type == AssetType.scene:
+        scene_name = export_data.get('scene')
+        scene = bpy.data.scenes[scene_name]
+        return scene.objects.keys()
+    return []
+
+
+def check_triangle_count(asset_type: AssetType, export_data: dict) -> Tuple[bool, str]:
+    """Check if triangle count is less than MAX_TRIANGLE_COUNT.
+
+    Parameters:
+        asset_type: type of asset that will be uploaded
+        export_data: dict containing objects to be uploaded info
+
+    Returns:
+        is_valid, message: if check passed and a report message
+    """
+    logging.info('Running triangle count...')
+    triangle_count = 0
+    object_list = _get_object_list(asset_type, export_data)
+    for object_name in object_list:
+        triangle_count = triangle_count + _get_triangle_count(object_name)
```

Comment:
Aqui pode ser `+=` em compensação haha

Reply:
Nesse caso o linter não deixou D:
```WPS519 Found implicit `sum()` call```

Reply:
:facepalm: 
Até daria para fazer sum com um list comprehension, mas isso é literalmente uma versão pior desse loop haha

Reply:
Tava fazendo uns testes para criar um issue no Wemake sobre isso e vi que o `sum` aceita generators.

Daria para fazer:
```suggestion
        triangle_count = sum(_get_triangle_count(object_name) for object_name in object_list)
```

Mas pode deixar assim, só achei interessante essa possibilidade